oioswift/common/middleware/container_hierarchy.py

```diff
@@ -393,6 +393,12 @@ def _path(self, obj_parts, is_mpu, sep=None):
                     except TypeError:
                         pass
             LOG.error("MPU fails to detect UploadId")
+
+        # CloudBerry backup: mitigate number of container created
+        if len(obj_parts) > 3:
+            if obj_parts[-1] + ':' == obj_parts[-3]:
+                return sep.join(obj_parts[:-3]), -2
+
```

Comment:
If I understand it correctly you are creating a container with the name path[:-3] and the object has the name path[-3:] ?

Reply:
Clouberry path for object versions is:

`[bucket]/[hostname]/[path_to_file]:/[version]`

Reply:
Nope, the PATH is really `[bucket]/[hostname]/[path_to_file]/[filename]:/[version]/[filename]`